drivers/sensor/hts221/hts221_trigger.c

```diff
@@ -110,8 +133,7 @@ int hts221_init_interrupt(struct device *dev)
 	}
 
 	gpio_pin_configure(drv_data->gpio, DT_INST_0_ST_HTS221_DRDY_GPIOS_PIN,
-			   GPIO_DIR_IN | GPIO_INT | GPIO_INT_EDGE |
-			   GPIO_INT_ACTIVE_HIGH | GPIO_INT_DEBOUNCE);
+			   GPIO_INPUT | GPIO_ACTIVE_HIGH | GPIO_INT_DEBOUNCE);
```

Comment:
This driver is using logical levels to configure the interrupts. This approach makes sense since we're talking about an external sensor. The signal `DRDY` which is used to trigger an interrupt can be routed on the board via logic that inverts it. Let's say the HTS221 sensor is in a different power domain and the `DRDY` signal is routed through an inverting level shifter. The driver doesn't know the specifics of the board design and needs to source this information from the DTS. We can't hard code here the signal as active high.

If we knew that the CPU and a sensor are connected directly via a wire then there would be no need to use logical levels. The driver knows exactly the model of the sensor so we could use physical levels.

Reply:
The HTS221 has a register that configures whether the signal it puts out is active high or active low.  The driver currently leaves that signal configured at its default, which is active high.  If the devicetree file specifies that the signal is active low, does that mean that the chip active-high signal is being inverted, or that the driver needs to configure the chip to active-low?

Whether it's physical or logical levels makes no difference.  The motivation for using logic levels is it simplifies the cases where we read the signal directly.

See also the thread I created on the slack channel related to this.